template/filter.c.erb

```diff
@@ -60,74 +55,108 @@ filter_message_iterator_next_processing_sending(
   btx_message_iterator_t *message_iterator_private_data =
       bt_self_message_iterator_get_data(self_message_iterator);
 
+  /* This should never append. When transition to sending, we check have some
+     message But to be safe, if such thing occur, we return AGAIN */
+  if (!message_iterator_private_data->queue) {
+    message_iterator_private_data->processing_state =
+        BTX_FILTER_PROCESSING_STATE_READING;
+    return BT_MESSAGE_ITERATOR_CLASS_NEXT_METHOD_STATUS_AGAIN;
+  }
+
```

Comment:
If this happens it is because we have an internal logic error. We should not hide it, we should maybe print and error message and return an error so we can fix it.